File: 1304-find-n-unique-integers-sum-up-to-zero/1304-find-n-unique-integers-sum-up-to-zero.c
int* sumZero(int n, int* returnSize){
    int *a = malloc(sizeof(int) * n);
    *returnSize = n;
    
    if(n % 2 == 1)
    {
        int j = -2;
        for(int i = 0; i < n; i++){
            if(i < n/2) a[i] = (-1)*(n/2) + i;
            else if(i == n/2) a[i] = 0;
            else{
                a[i] = (-1)*a[i+j];
                j -= 2;
            }        
        }
    }
    else
    {
        int j = -1;
        for(int i = 0; i < n; i++){
            if(i < n/2) a[i] = (-1)*(n/2) + i;
            else{
                a[i] = (-1) * a[i+j];
                j -= 2;
            } 
        }
    }
    return a;
}

File: 1304-find-n-unique-integers-sum-up-to-zero/1304-find-n-unique-integers-sum-up-to-zero.java
class Solution {
    public int[] sumZero(int n) {
        int[] arr = new int[n];
        int val;
        
        if(n % 2 == 0){
            val = -n + 1;
            for(int i = 0; i < n; i++){
                arr[i] = val;
                val += 2;
            }
        }
        else{
            val = -n/2;
            for(int i = 0; i < n; i++){
                arr[i] = val++;
            }
        }    
        return arr;
    }
}